terraform/provider/resource_teleport_cluster_audit_config.go

```diff
@@ -59,8 +63,9 @@ func resourceAuditConfigCreate(ctx context.Context, d *schema.ResourceData, m in
 		return diagFromErr(err)
 	}
 
-	err = c.SetClusterAuditConfig(ctx, &n)
-	if err != nil {
+	// Linter generates false positive here because the API always returns error (see comment above)
```

Comment:
If it always returns an error, why are we doing it?

Reply:
That's the legacy. As I mentioned above, those API methods were disabled some time ago (Teleport PR#7465). They always return NotImplementedError, and that causes linter failure: it detects that the result of the method call is always error, hence, there's no reason to perform `if err != nil`. I can remove this entity from Terraform if we have no plans to revive it. And I think that API methods should have been removed, but for some reason they are still there - and that's why I chose to keep the resource on Terraform side as well.

Reply:
`AuditConfig` was not meant to be dynamically configurable, so they should be removed from the Terraform provider. - https://github.com/gravitational/teleport/pull/7465. 

These unimplemented methods are used for so that the client fits internal interfaces, but there's no reason they should be exposed in our public API. I'll make a PR to fix it.

Reply:
OK, so we will merge this as-is and remopve these calls in a subsequent PR. Everyone OK with that?